apps/natdex/natdex.star

```diff
@@ -0,0 +1,165 @@
+"""
+Applet: National Pokedex
+Summary: Display a random Pokemon from Gen I - VII
+Description: Display a random Pokemon from your region of choice
+Author: Lauren Kopac
+"""
+
+load("cache.star", "cache")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("re.star", "re")
+
+POKEAPI_URL = "https://pokeapi.co/api/v2/pokemon/{}"
+REGIONAL_DEX_ID = "regional_dex_code"
+CACHE_TTL_SECONDS = 3600 * 24 * 7  # 7 days in seconds.
+
+def get_regions():
+    regions = ["National", "Kanto", "Johto", "Hoenn", "Sinnoh", "Unova", "Kalos", "Alola"]
+    region_options = []
+    for x in regions:
+        region_options.append(
+            schema.Option(
+                display = x,
+                value = x,
+            ),
+        )
+    return region_options
+
+def get_schema():
+    regions = get_regions()
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Dropdown(
+                id = REGIONAL_DEX_ID,
+                name = "Regional Pokedex",
+                desc = "Which Pokedex do you want to pull from?",
+                icon = "book",
+                default = regions[0].value,
+                options = regions,
+            ),
+        ],
+    )
+
+def main(config):
+    MIN = "1"
+    MAX = "809"
+    dex_region = config.get(REGIONAL_DEX_ID)
+    if dex_region == "National":
+        pass
+    elif dex_region == "Kanto":
+        MIN = "1"
+        MAX = "151"
+    elif dex_region == "Johto":
+        MIN = "152"
+        MAX = "251"
+    elif dex_region == "Hoenn":
+        MIN = "252"
+        MAX = "386"
+    elif dex_region == "Sinnoh":
+        MIN = "387"
+        MAX = "493"
+    elif dex_region == "Unova":
+        MIN = "494"
+        MAX = "649"
+    elif dex_region == "Kalos":
+        MIN = "650"
+        MAX = "721"
+    elif dex_region == "Alola":
+        MIN = "722"
+        MAX = "809"
+    else:
+        pass
+    resp = http.get("https://www.random.org/integers/?num=1&min=" + MIN + "&max=" + MAX + "&col=1&base=10&format=plain&rnd=new1")
```

Comment:
We added the `random` module after you created this PR 🤦  but it could eliminate this HTTP request